core/logger/pyroscope.go

```diff
@@ -5,14 +5,13 @@ import (
 
 	"github.com/pyroscope-io/client/pyroscope"
 
+	"github.com/smartcontractkit/chainlink/v2/core/config"
 	"github.com/smartcontractkit/chainlink/v2/core/static"
 )
 
 // PyroscopeConfig represents the expected configuration for Pyroscope to properly work
 type PyroscopeConfig interface {
-	PyroscopeServerAddress() string
-	PyroscopeAuthToken() string
-	PyroscopeEnvironment() string
+	Pyroscope() config.Pyroscope
```

Comment:
@george-dorin Can we pass this in separately here as per option 1?

Reply:
Ah, yep that makes sense!

Reply:
The naming is a bit odd now though. Should `PyroscopeConfig` be called something else?

Reply:
@jmank88 how about `PyroscopeProfileConfig`?

Reply:
SGTM